core/templates/expressions/expression-syntax-tree.service.ts

```diff
@@ -16,417 +16,445 @@
-          return numericArgs[0] <= numericArgs[1];
-        },
-        getType: function(args) {
-          verifyNumArgs(args, 2);
-          _verifyArgTypesMatchExpectedType(args, PARAMETER_TYPES.REAL);
-          return PARAMETER_TYPES.UNICODE_STRING;
-        }
+      getType: (args: Array<any>): string => {
+        this.verifyNumArgs(args, 2);
+        this._verifyArgTypesMatchExpectedType(args,
+          AppConstants.PARAMETER_TYPES.REAL);
+        return AppConstants.PARAMETER_TYPES.REAL;
+      }
+    },
+    '<=': {
+      eval: (args: Array<any>): boolean => {
+        this.verifyNumArgs(args, 2);
+        var numericArgs = this._coerceAllArgsToNumber(args);
+        return numericArgs[0] <= numericArgs[1];
       },
-      '>=': {
-        eval: function(args) {
-          verifyNumArgs(args, 2);
-          var numericArgs = _coerceAllArgsToNumber(args);
-          return numericArgs[0] >= numericArgs[1];
-        },
-        getType: function(args) {
-          verifyNumArgs(args, 2);
-          _verifyArgTypesMatchExpectedType(args, PARAMETER_TYPES.REAL);
-          return PARAMETER_TYPES.UNICODE_STRING;
-        }
+      getType: (args: Array<any>): string => {
+        this.verifyNumArgs(args, 2);
+        this._verifyArgTypesMatchExpectedType(args,
+          AppConstants.PARAMETER_TYPES.REAL);
+        return AppConstants.PARAMETER_TYPES.UNICODE_STRING;
+      }
+    },
+    '>=': {
+      eval: (args: Array<any>): boolean => {
+        this.verifyNumArgs(args, 2);
+        var numericArgs = this._coerceAllArgsToNumber(args);
+        return numericArgs[0] >= numericArgs[1];
       },
-      '<': {
-        eval: function(args) {
-          verifyNumArgs(args, 2);
-          var numericArgs = _coerceAllArgsToNumber(args);
-          return numericArgs[0] < numericArgs[1];
-        },
-        getType: function(args) {
-          verifyNumArgs(args, 2);
-          _verifyArgTypesMatchExpectedType(args, PARAMETER_TYPES.REAL);
-          return PARAMETER_TYPES.UNICODE_STRING;
-        }
+      getType: (args: Array<any>): string => {
+        this.verifyNumArgs(args, 2);
+        this._verifyArgTypesMatchExpectedType(args,
+          AppConstants.PARAMETER_TYPES.REAL);
+        return AppConstants.PARAMETER_TYPES.UNICODE_STRING;
+      }
+    },
+    '<': {
+      eval: (args: Array<any>): boolean => {
+        this.verifyNumArgs(args, 2);
+        var numericArgs = this._coerceAllArgsToNumber(args);
+        return numericArgs[0] < numericArgs[1];
       },
-      '>': {
-        eval: function(args) {
-          verifyNumArgs(args, 2);
-          var numericArgs = _coerceAllArgsToNumber(args);
-          return numericArgs[0] > numericArgs[1];
-        },
-        getType: function(args) {
-          verifyNumArgs(args, 2);
-          _verifyArgTypesMatchExpectedType(args, PARAMETER_TYPES.REAL);
-          return PARAMETER_TYPES.UNICODE_STRING;
-        }
+      getType: (args: Array<any>): string => {
```

Comment:
Is this really any or do we don't know the exact type? Ditto here and below.

Reply:
You are right, I just console.log them during the test and noticed the `args` array contains strings. I should've consoled them in the first place. 😓

Reply:
@bansalnitish It's more complicated then I thought. Here's what happens:
If I set `system.pow.eval`s args to `Array<string>`
<img width="902" alt="Screenshot 2020-03-24 at 11 58 45" src="https://user-images.githubusercontent.com/11369764/77418194-da33c700-6dc6-11ea-8df0-daa16c97720f.png">
`Math.pow`s first argument will throw this error. If I would set it to `Array<number>` then `this.verifyNumArgs`s first parameter would throw a similar error and because I know from the `console.log` you get strings so I was forced to set to `Array<any>`. 😲

Your thoughts on this?

Reply:
For these can we set them to `Array<number|string>` then which is far better than `Array<any>`?

Also, aren't these two different functions one uses `Array<string>` the other uses `Array<number>` (didn't go completely through the codebase, first suggestion would anyway work)?